backstop_data/backstop_test.js

```diff
@@ -0,0 +1,135 @@
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const scenarioConfigs = require('./backstop_scenarios');
+const scenarios = [];
+const excludedScenarios = new Set(['@spectrum-css/circleloader', '@spectrum-css/coachmark', 'index', 'getting-started']);
+const themes = new Set().add('light');
+const scales = new Set().add('medium');
+const packageNameSet = new Set();
+const packageDependentMap = new Map(require('./packageDependentMap'));
+
+let isDocker = false;
+let env = 'local';
+let host = 'host.docker.internal';
+let report = 'ci';
+let captureLimit = 1;
+
+// Shared scenario configuration
+const baseScenarioConfig = {
+  referenceUrl: '',
+  readyEvent: '',
+  readySelector: '',
+  delay: 0,
+  hideSelectors: [],
+  removeSelectors: [],
+  hoverSelector: '',
+  clickSelector: '',
+  postInteractionWait: 0,
+  selectors: ['.spectrum-CSSExample-example'],
+  selectorExpansion: true,
+  expect: 0,
+  misMatchThreshold: 0.1,
+  requireSameDimensions: true
+};
+
+const [...rest] = process.argv.slice(3); // Exclude 'node', 'backstop' and backstop sub command like 'test/approve'
+
+// Process the arguments
+rest.forEach(argv => {
+  if (argv === '--moby') {
+    isDocker = true;
+  } else if (argv.startsWith('--env=')) {
+    env = argv.slice('--env='.length);
+  } else if (argv.startsWith('themes=')) {
+    themes.clear();
+    argv.slice('themes='.length).split(',').forEach(t => themes.add(t));
+  } else if (argv.startsWith('scales=')) {
+    scales.clear();
+    argv.slice('scales='.length).split(',').forEach(s => scales.add(s));
+  } else if (!argv.startsWith('--')) {
+    packageNameSet.add(`@spectrum-css/${argv}`);
+    packageDependentMap.get(`@spectrum-css/${argv}`).forEach(i => packageNameSet.add(i));
+  }
+});
+
+// Identify the runtime environment and setup hostname, etc
+if (isDocker === true) {
+  host = (env === 'local' ? 'host.docker.internal' : '127.0.0.1');
+} else {
+  host = '127.0.0.1';
+}
+
+if (env === 'local') {
+  report = 'browser';
+  captureLimit = 5;
+} else {
+  report = 'CI';
+}
+
+// Generate vr testing scenarios
+let testingScenarios = [];
+
+if (packageNameSet.size > 0) {
+  scenarioConfigs.forEach(s => {
+    if(packageNameSet.has(s.package)) {
+      testingScenarios.push(s);
+    }
+  });
+  testingScenarios = testingScenarios.filter(i => !(excludedScenarios.has(i.package) || excludedScenarios.has(i.label)));
+} else {
+  testingScenarios = scenarioConfigs.filter(i => !(excludedScenarios.has(i.package) || excludedScenarios.has(i.label)));
+}
+
+testingScenarios.map(specificScenarioConfig => {
+  themes.forEach(t => {
+    scales.forEach(s => {
+      const config = {...specificScenarioConfig};
+      config.label = `${config.label}-${t}-${s}`;
+      config.url = `http://${host}:3000/docs/${config.url}`;
```

Comment:
I assume this is the internal port inside the docker image? 
A lot of other projects here use that port for serving files. Maybe it could be configurable?

Reply:
hmm...The `http://${host}:3000/` is determined by how I start the dev server. I hard code `3000` port in https://github.com/adobe/spectrum-css/pull/271/files#diff-354f30a63fb0907d4ad57269548329e3.

Reply:
I think that can be configured in BrowserSync : https://www.browsersync.io/docs/options#option-port

Reply:
Yes, but `gulp dev` task did not open this option for user to configure it. If we want to make it configurable, we should do it together. It is out of the scope.

Reply:
ok, next time then